# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLUniformLocation.h
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGLUniformLocation.h
--- a/dom/canvas/WebGLUniformLocation.h	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGLUniformLocation.h	Wed Jan 08 22:19:14 2020 +0000
@@ -8,9 +8,7 @@
 
 #include "GLDefs.h"
 #include "mozilla/WeakPtr.h"
-#include "nsCycleCollectionParticipant.h"  // NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_NATIVE_CLASS
 #include "nsISupportsImpl.h"  // NS_INLINE_DECL_CYCLE_COLLECTING_NATIVE_REFCOUNTING
-#include "nsWrapperCache.h"
 
 #include "WebGLObjectModel.h"
 
@@ -26,16 +24,10 @@
 struct UniformInfo;
 }  // namespace webgl
 
-class WebGLUniformLocation final : public nsWrapperCache,
-                                   public WebGLContextBoundObject {
+class WebGLUniformLocation final
+    : public WebGLContextBoundObject<WebGLUniformLocation> {
  public:
-  NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_NATIVE_CLASS(WebGLUniformLocation)
-  NS_INLINE_DECL_CYCLE_COLLECTING_NATIVE_REFCOUNTING(WebGLUniformLocation)
-
-  virtual JSObject* WrapObject(JSContext* js,
-                               JS::Handle<JSObject*> givenProto) override;
-
-  WebGLContext* GetParentObject() const { return mContext; }
+  NS_INLINE_DECL_REFCOUNTING(WebGLUniformLocation)
 
   //////
 
@@ -56,7 +48,7 @@
   bool ValidateArrayLength(uint8_t setterElemSize,
                            size_t setterArraySize) const;
 
-  JS::Value GetUniform(JSContext* js) const;
+  MaybeWebGLVariant GetUniform() const;
 
   // Needed for certain helper functions like ValidateObject.
   // `WebGLUniformLocation`s can't be 'Deleted' in the WebGL sense.